enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -597,66 +842,94 @@ Below are the example static manifests used for creating required resources for
+               - --root-ca-file=
+               - --serving-certificate-dns-names=cert-manager-istio-csr.cert-manager.svc
+               - --serving-certificate-duration=1h
+               - --trust-domain=cluster.local
+               - --cluster-id=Kubernetes
+               - --max-client-certificate-duration=1h
+               - --serving-address=0.0.0.0:6443
+               - --serving-certificate-key-size=2048
+               - --serving-signature-algorithm=RSA
+               - --enable-client-cert-authenticator=false
+               - --leader-election-namespace=istio-system
+               - --disable-kubernetes-client-rate-limiter=false
+               - --runtime-issuance-config-map-name=
+               - --runtime-issuance-config-map-namespace=cert-manager
+               - --istiod-cert-enabled=false
+               - --istiod-cert-name=istiod-dynamic
+               - --istiod-cert-namespace=istio-system
+               - --istiod-cert-duration=1h
+               - --istiod-cert-renew-before=30m
+               - --istiod-cert-key-algorithm=RSA
+               - --istiod-cert-key-size=2048
+               - --istiod-cert-additional-dns-names=
+               - --istiod-cert-istio-revisions=default
+             resources: {}
+             securityContext:
+               allowPrivilegeEscalation: false
+               capabilities:
+                 drop:
+                   - ALL
+               readOnlyRootFilesystem: true
+               runAsNonRoot: true
+               seccompProfile:
+                 type: RuntimeDefault
    ```
 
 ### Risks and Mitigations
 
-An OpenShift administrator configuring `istiocsr.operator.openshift.io` CR object could configure insecure certificate
+- An OpenShift administrator configuring `istiocsr.operator.openshift.io` CR object could configure insecure certificate
 signature algorithm, certificate key size or certificate validity to be too long which could cause vulnerability.
   - These configurations could be validated and can be overridden with default values.
+- `istio-csr` agents creates the configmap `istio-ca-root-cert` with the CA certificate used for signing the istio server
+  certificates. And istio also creates the same configmap when it's configured to obtain certificates from `istio-csr`. Due
+  to both `istio` and `istio-csr` creating the configmap with same name for same purpose below issues can occur
+  - When multiple istio instance are installed, and not all configured to use `istio-csr` for certificate needs, conflict
+    will happen with both trying to update the CA certificate configmaps.
+  - Mitigation is to make use the namespace filtering `spec.istioCSRConfig.istioDataPlaneNamespaceSelector` in `istiocsrs.operator.openshift.io`
+    and the equivalent in `istio` to avoid the conflict. When using `spec.istioCSRConfig.istioDataPlaneNamespaceSelector`
+    one should be mindful to label the namespaces to not overlap with multiple instances of istio.
```

Comment:
Another limitation can also be called out

```
When adding or changing the `spec.istioCSRConfig.istioDataPlaneNamespaceSelector`, existing istio-ca-root-cert ConfigMaps in non-matching namespaces are not automatically removed and require manual cleanup.
```

Reply:
suggestion incorporated.